src/OpenLoco/Windows/PromptBrowseWindow.cpp

```diff
@@ -758,99 +679,63 @@ namespace OpenLoco::Ui::Windows::PromptBrowse
-                        if (f.is_regular_file())
-                        {
-                            auto extension = f.path().extension().u8string();
-                            if (!Utility::iequals(extension, filterExtension))
-                            {
-                                continue;
-                            }
-                        }
-                        else if (!isDirectory) // Only list directories or normal files
-                        {
+                        auto extension = file.path().extension().u8string();
+                        if (!Utility::iequals(extension, filterExtension))
                             continue;
-                        }
-                        auto name = f.path().stem().u8string();
-                        _newFiles.emplace_back(name, isDirectory);
                     }
-                }
-                catch (const fs::filesystem_error& err)
-                {
-                    Console::error("Invalid directory or file: %s", err.what());
+
+                    _files.emplace_back(file);
                 }
             }
+            catch (const fs::filesystem_error& err)
+            {
+                Console::error("Invalid directory or file: %s", err.what());
+            }
         }
 
-        std::sort(_newFiles.begin(), _newFiles.end(), [](const FileEntry& a, const FileEntry& b) -> bool {
-            if (!a.isDirectory() && b.isDirectory())
+        std::sort(_files.begin(), _files.end(), [](const fs::directory_entry& a, const fs::directory_entry& b) -> bool {
+            if (!a.is_directory() && b.is_directory())
                 return false;
-            if (a.isDirectory() && !b.isDirectory())
+            if (a.is_directory() && !b.is_directory())
                 return true;
-            return a.getName() < b.getName();
+            return a.path().stem() < b.path().stem();
         });
-
-        _numFiles = (int16_t)_newFiles.size();
-        _files = _newFiles.data();
     }
 
     // 0x00446E2F
     static void upOneLevel()
     {
-        char* ptr = _directory;
-        while (*ptr != '\0')
-            ptr++;
-
-        ptr--;
-        if (*ptr == fs::path::preferred_separator)
-            ptr--;
-
-        while (ptr != _directory && *ptr != fs::path::preferred_separator)
-            ptr--;
-
-        if (*ptr == fs::path::preferred_separator)
-            ptr++;
-
-        *ptr = '\0';
-
+        _currentDirectory = _currentDirectory.parent_path().parent_path() / "";
         refreshDirectoryList();
     }
 
     // 0x00446E62
     static void appendDirectory(const char* to_append)
     {
-        char* dst = _directory;
-        while (*dst != '\0')
-            dst++;
-
-        const char* src = to_append;
-        while (*src != '\0')
-        {
-            *dst++ = *src++;
-        }
-
-        *dst++ = fs::path::preferred_separator;
-        *dst = '\0';
-
+        _currentDirectory = _currentDirectory / to_append / "";
```

Comment:
modify function to take a path then you dont need to mess about converting it to a u8.

Reply:
Hmm. Just a filename doesn't really constitute a full path though. If instead we pass a full path, perhaps we should just change the function to `changeDirectory` instead, then?

Reply:
That's probably a good option as well. I was just looking at the type that stem() returns and it's path. So assumed you can compose with that.